### DISCUSSION TITLE
Feature Request: Last Zone Opened Attribute

### DISCUSSION THREAD

**Discussion by PablaV** [ufodone/envisalink_new#57] [Ideas]
I have been trying to set up a notification that tells me what zone tripped the alarm. The problem I am running into is that it's pretty difficult to keep track of what zone was tripped since there could be entry delay or the zone could've been closed before the alarm actually goes off. Would be nice to have an attribute that states whatever last zone was opened so that it can be used in automations in the event of an alarm.

**Comment by ufodone** [comment]
Are you looking for the last zone faulted before the alarm was tripped or the last zone faulted?  I think those two might be different since theoretically I'd imagine it's possible for other zones to fault while the alarm is active.

**Reply by PablaV** [reply]
That's a fair point, probably the last zone faulted is what would be most useful.

**Comment by ufodone** [comment]
I was thinking about this a bit more and think it might be better to just implement it with an automation and store the state in something like a text helper.  Something like:

```- id: '1701050591467'
  alias: Track Last Zone Faulted
  description: ''
  trigger:
  - platform: state
    entity_id:
    - binary_sensor.alarm_zone_1
    - binary_sensor.alarm_zone_2
    - binary_sensor.alarm_zone_3
    - binary_sensor.alarm_zone_4
    - binary_sensor.alarm_zone_5
    - binary_sensor.alarm_zone_6
    - binary_sensor.alarm_zone_7
    - binary_sensor.alarm_zone_8
    to: 'on'
  condition: []
  action:
  - service: input_text.set_value
    data:
      value: '{{trigger.entity_id}}'
    target:
      entity_id: input_text.last_zone_faulted
  mode: queued
  max: 10
```

That would also allow you to tune it to your needs.  E.g., use a condition to only update it if the zone is faulted when the alarm is armed.

**Comment by chelming** [comment]
Doing this as a template should work. I've edited this post like 4 times now but here's a generic solution that should work for everyone:
```jinja2
{{
  expand(integration_entities('envisalink_new')) |
  selectattr('attributes.last_tripped_time','defined') |
  rejectattr('attributes.last_tripped_time','none') |
  sort(attribute='attributes.last_tripped_time', reverse=true) |
  map(attribute='name') |
  first 
}}

```
This returns the friendly name. If you want entity id just change that map line and replace name with entity_id

**Reply by tivo65** [reply]
thanks chelming - this is exactly what I was trying to do.  I threw this into my templates.yaml file

`
- sensor:
    - name: "latest_alarm_tripped_sensor"
      state: >
        {{  
          expand(integration_entities('envisalink_new')) |  
          selectattr('attributes.last_tripped_time','defined') |  
          rejectattr('attributes.last_tripped_time','none') |  
          sort(attribute='attributes.last_tripped_time', reverse=true) |  
          map(attribute='name') |  
          first  
        }}
      attributes:
        last_tripped_time: >
          {{  
            expand(integration_entities('envisalink_new')) |  
            selectattr('attributes.last_tripped_time','defined') |  
            rejectattr('attributes.last_tripped_time','none') |  
            sort(attribute='attributes.last_tripped_time', reverse=true) |  
            map(attribute='attributes.last_tripped_time') |  
            first  
          }}
`
